core/src/commonTest/kotlin/AutoIncrementKeyObjectStore.kt

```diff
@@ -1,30 +1,36 @@
-package com.juul.indexeddb
-
+import com.juul.indexeddb.external.IDBKey
+import com.juul.indexeddb.external.JsAny
+import com.juul.indexeddb.jso
+import com.juul.indexeddb.unsafeCast
+import kotlinx.coroutines.test.runTest
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
-class AutoIncrementKeyObjectStore {
+private external interface User : JsAny {
+    var username: String
+}
 
+class AutoIncrementKeyObjectStore {
     @Test
     fun simpleReadWrite() = runTest {
         val database = openDatabase("auto-increment-keys", 1) { database, oldVersion, newVersion ->
             if (oldVersion < 1) {
                 database.createObjectStore("users", AutoIncrement)
             }
         }
-        onCleanup {
-            database.close()
-            deleteDatabase("auto-increment-keys")
-        }
 
         val id = database.writeTransaction("users") {
-            objectStore("users").add(jso { username = "Username" }) as Double
+            objectStore("users").add(jso<User> { username = "Username" })
         }
 
         val user = database.transaction("users") {
             objectStore("users")
-                .get(Key(id))
+                .get(IDBKey(id))
+                ?.unsafeCast<User>()
         }
-        assertEquals("Username", user.username)
+        assertEquals("Username", user?.username)
+
+        database.close()
+        deleteDatabase("auto-increment-keys")
```

Comment:
I'm not opposed to losing the test wrapper if there was a real reason it had to go (I'm guessing it didn't play nice with wasm?), but if it's gone there needs to be a `try ... finally` for these cleanup lines. Otherwise a failing test leaves some garbage behind that can make future test runs (which should be successful) fail. Same goes for the other tests

Reply:
There was an issue with Wasm, although I have a few ideas about how to make it work. 

But worst case scenario I will add the `try...finally`